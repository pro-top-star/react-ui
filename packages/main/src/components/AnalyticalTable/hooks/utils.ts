export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
