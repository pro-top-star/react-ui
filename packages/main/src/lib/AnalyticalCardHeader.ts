import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
