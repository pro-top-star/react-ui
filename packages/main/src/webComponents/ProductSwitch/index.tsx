import UI5ProductSwitch from '@ui5/webcomponents-fiori/dist/ProductSwitch';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import React, { FC, ReactNode } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface ProductSwitchPropTypes extends WithWebComponentPropTypes {
  children?: ReactNode | ReactNode[]; // @generated
}

/**
 * <code>import { ProductSwitch } from '@ui5/webcomponents-react/lib/ProductSwitch';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/ProductSwitch" target="_blank">UI5 Web Components Playground</a>
 */
const ProductSwitch: FC<ProductSwitchPropTypes> = withWebComponent<ProductSwitchPropTypes>(UI5ProductSwitch);

ProductSwitch.displayName = 'ProductSwitch';

export { ProductSwitch };
