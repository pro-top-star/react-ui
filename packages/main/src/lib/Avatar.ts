import { Avatar } from '../webComponents/Avatar';

export { Avatar };
