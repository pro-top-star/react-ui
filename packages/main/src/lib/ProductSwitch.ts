import { ProductSwitch } from '../webComponents/ProductSwitch';

export { ProductSwitch };
