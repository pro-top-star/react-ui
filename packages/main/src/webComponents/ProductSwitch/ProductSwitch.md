[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/ProductSwitch)
```jsx
import { ProductSwitch } from '@ui5/webcomponents-react/lib/ProductSwitch';
```
