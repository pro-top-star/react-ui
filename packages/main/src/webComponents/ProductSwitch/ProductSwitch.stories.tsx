import '@ui5/webcomponents-icons/dist/icons/business-objects-experience';
import '@ui5/webcomponents-icons/dist/icons/contacts';
import '@ui5/webcomponents-icons/dist/icons/flight';
import '@ui5/webcomponents-icons/dist/icons/home';
import { ProductSwitch } from '@ui5/webcomponents-react/lib/ProductSwitch';
import { ProductSwitchItem } from '@ui5/webcomponents-react/lib/ProductSwitchItem';
import React from 'react';

export default {
  title: 'UI5 Web Components / ProductSwitch',
  component: ProductSwitch,
  parameters: {
    subcomponents: { ProductSwitchItem }
  }
};

export const generatedDefaultStory = () => (
  <ProductSwitch>
    <ProductSwitchItem heading="Home" subtitle="Central Home" icon="home" />
    <ProductSwitchItem heading="Analytics Cloud" subtitle="Analytics Cloud" icon="business-objects-experience" />
    <ProductSwitchItem heading="Catalog" subtitle="Ariba" icon="contacts" />
    <ProductSwitchItem heading="Travel & Expense" subtitle="Concur" icon="flight" />
  </ProductSwitch>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
