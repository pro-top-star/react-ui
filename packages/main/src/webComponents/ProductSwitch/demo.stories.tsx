import { text } from '@storybook/addon-knobs';
import { ProductSwitch } from '@ui5/webcomponents-react/lib/ProductSwitch';
import { ProductSwitchItem } from '@ui5/webcomponents-react/lib/ProductSwitchItem';
import React from 'react';
import notes from './ProductSwitch.md';

export default {
  title: 'UI5 Web Components / ProductSwitch',
  component: ProductSwitch,
  parameters: {
    notes
  }
};

export const generatedDefaultStory = () => (
  <ProductSwitch>
    <ProductSwitchItem
      heading={text('heading', 'SAP')}
      subtitle={text('subtitle', 'Website')}
      icon="add"
      targetSrc={text('targetSrc', 'https://sap.com')}
      target={text('target', '_blank')}
    />
    <ProductSwitchItem
      heading="UI5 Web Components"
      targetSrc="https://github.com/SAP/ui5-webcomponents"
      target="_blank"
      icon="add"
    />
    <ProductSwitchItem
      heading="UI5 Web Components"
      subtitle="for React"
      targetSrc="https://github.com/SAP/ui5-webcomponents-react"
      target="_blank"
      icon="add"
    />
  </ProductSwitch>
);

generatedDefaultStory.story = {
  name: 'Generated Default Story'
};
