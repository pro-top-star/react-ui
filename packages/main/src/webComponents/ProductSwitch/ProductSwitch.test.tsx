import { mountThemedComponent } from '@shared/tests/utils';
import { ProductSwitch } from '@ui5/webcomponents-react/lib/ProductSwitch';
import React from 'react';

describe('ProductSwitch', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<ProductSwitch />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
