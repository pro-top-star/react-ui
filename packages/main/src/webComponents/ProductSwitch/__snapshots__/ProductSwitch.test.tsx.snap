// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSwitch Basic Test (generated) 1`] = `<ui5-product-switch />`;
